pvacseq/lib/convert_vcf.py

```diff
@@ -3,289 +3,293 @@
 import csv
 import sys
 import re
+from abc import ABCMeta
 
-def parse_bam_readcount_file(bam_readcount_file):
-    coverage_tsv_reader = csv.reader(bam_readcount_file, delimiter='\t')
-    coverage = {}
-    for row in coverage_tsv_reader:
-        chromosome     = row[0]
-        position       = row[1]
-        reference_base = row[2].upper()
-        depth          = row[3]
-        brct           = row[4:]
-        if chromosome not in coverage:
-            coverage[chromosome] = {}
-        if position not in coverage[chromosome]:
-            coverage[chromosome][position] = {}
-        coverage[chromosome][position][reference_base] = brct
-    return coverage
+class ConvertVcf(metaclass=ABCMeta):
+    def __init__(self, **kwargs):
+        self.input_file                  = kwargs['input_file']
+        self.output_file                 = kwargs['output_file']
+        self.gene_expn_file              = kwargs['gene_expn_file']
+        self.transcript_expn_file        = kwargs['transcript_expn_file']
+        self.normal_snvs_coverage_file   = kwargs['normal_snvs_coverage_file']
+        self.normal_indels_coverage_file = kwargs['normal_indels_coverage_file']
+        self.tdna_snvs_coverage_file     = kwargs['tdna_snvs_coverage_file']
+        self.tdna_indels_coverage_file   = kwargs['tdna_indels_coverage_file']
+        self.trna_snvs_coverage_file     = kwargs['trna_snvs_coverage_file']
+        self.trna_indels_coverage_file   = kwargs['trna_indels_coverage_file']
 
-def parse_brct_field(brct_entry):
-    parsed_brct = {}
-    for brct in brct_entry:
-        (base, count, rest) = brct.split(':', 2)
-        parsed_brct[base.upper()] = count
-    return parsed_brct
+    def parse_bam_readcount_file(self, bam_readcount_file):
+        reader              = open(bam_readcount_file, 'r')
```

Comment:
I went through this file and replaced these for the `with` version in a few places.